### DISCUSSION TITLE
Best way to have a globally accessible experimental env

### DISCUSSION THREAD

**Discussion by foxale** [python-poetry/poetry#6811] [Q&A]
My current setup is:
- `pyenv` for multiple python versions, but all clean (with no packages installed)
- `poetry` for project environments managed via `pipx`

This is a perfect setup for working on multiple projects, as I have separate python versions, environments, etc.

However, many times I have a situation that I'm in a random directory and need to quickly pull up python interactive prompt to verify some random though. Usually a bare python is not enough, I need it to have a bunch of most often used packages pre-installed (e.g. pandas, numpy).

In other words, **I need to have a globally accessible python environment for experimenting with random stuff**, not related to any particular project. 

What's the best way to achieve this, without cluttering the pyenv python interpreters with random packages?

**Comment by neersighted** [comment]
I highly suggest use of `pipx` to manage Poetry and other Python CLI tools that are used in a global and not project-specific context: https://pypa.github.io/pipx/

**Reply by foxale** [reply]
I use `pipx` to manage Poetry, but how is that related to the question?

**Reply by foxale** [reply]
I edited the post to make it more clear what is my goal :)

**Comment by neersighted** [comment]
Ah, I see you are referring to importable modules and not tools with entrypoints. I think the answer to that question is the same as any other tool -- install things into one or more sandbox/pet virtual envs you keep in a common location.

e.g.

```sh
python3 -m venv ~/.virtualenvs/sandbox
source ~/.virtualenvs/sandbox/activate
```

There are common tools to support this workflow, like pyenv-virtualenv and virtualenvwrapper.
